plasma-mode.md

```diff
@@ -0,0 +1,203 @@
+
+
+### Alternatives
+
+We considered and rejected making the op-node aware of each DA Layer's commitment format.
+This would allow the op-node to be stricter about validating which DA layers are being used and relies less on the DA Server.
+The downside is that this requires each DA Layer to integrate with the specs and op-node source.
+
+
+## Data Availability Challenge
+
+The Data Availability Challenge ensures that nodes are able to fetch the data corresponding to the commitment posted on L1.
+If the data is not available, nodes should not attempt to use the commitment.
+If the data is only made available to a small portion of nodes, it should either be made available to all nodes by posting the data to L1 or the commitment should be rejected.
+Ensuring that nodes have a consistent view of the data is important to ensure that nodes derive the same chain and the fault proofs function correctly.
+
+Each DA Layer will have to implement logic on L1 to validate their commitments. This logic will emit a specific set of events that the op-node will read.
+These events then get adapted into a golang interface that the derivation pipeline uses to manage L2 reorgs & finality.
+
+### Data Availability Challenge Smart Contract Specification
+
+#### V1: Optimisic Bridge
+The DAC SC will emit the following event
+```solidity
+enum ChallengeStatus {
+    Uninitialized, // Never emitted
+    Active,        // Emitted when a challenge is initiated
+    Resolved,      // Emitted when a challenge is resolved. This means the data has been made available on L1.
+    Expired        // Never emitted
+}
+
+event ChallengeStatusChanged(bytes indexed challengedCommitment, uint256 indexed challengedBlockNumber, ChallengeStatus status);
+```
+
+The flow is that after a commitment is posted to L1, it can be challenged within `challenge_window` blocks.
+If it is challenged an event with the `Active` status is emitted.
+If the challenge has not been resolved within `resolve_window` blocks, the challenge is presumed to suceed and the commitment is invalid.
+If the challenge is resolved within `resolve_window`, an event with the `Resolved` status is emitted and the commitment is valid.
+If a commitment is never challenged within `challenge_window` blocks, it is valid.
+
+This works essentially as an optimistic bridge. The commitment is assumed to be valid until someone claims that it is not.
+The validity is checked by a third party smart contract which understands the commitment format.
+
+
+#### V2: ZK Bridge
```

Comment:
Can you give a more thorough example into how this will work with blobstream?

Reply:
I haven't really scoped this out yet. I'd assume that the Celestia DAC would reach out to the blobstream contract and call https://github.com/succinctlabs/blobstreamx/blob/aac0842f17056e5343f66de7df44020c1637e8b7/contracts/src/BlobstreamX.sol#L252-L281

cc @tuxcanfly will this interface work with blobstream

Reply:
Yes, given a commitment and it's inclusion proof blobstream can resolve the challenge by returning whether the attested data is available on celestia.

However, we will need the inclusion proof to be submitted to L1.